src/main/java/org/openrewrite/quarkus/AddQuarkusProperty.java

```diff
@@ -0,0 +1,169 @@
+    @Nullable
+    String comment;
+
+    @Option(displayName = "Profile",
+            description = "The profile to add the property to. If not specified, the property will be added to the default profile.",
+            required = false,
+            example = "dev")
+    @Nullable
+    String profile;
+
+    @Option(displayName = "Optional list of file path matcher",
+            description = "Each value in this list represents a glob expression that is used to match which files will " +
+                          "be modified. If this value is not present, this recipe will query the execution context for " +
+                          "reasonable defaults. (\"**/application.yml\", \"**/application.yaml\", " +
+                          "\"**/application.properties\" and \"**/META-INF/microprofile-config.properties\".",
+            required = false,
+            example = "[\"**/application.yaml\"]")
+    @Nullable
+    List<String> pathExpressions;
+
+    @Override
+    public String getDisplayName() {
+        return "Add a Quarkus configuration property";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Add a Quarkus configuration property to an existing configuration file if it does not already exist in that file.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public boolean isAcceptable(SourceFile sourceFile, ExecutionContext ctx) {
+                return sourceFile instanceof Yaml.Documents || sourceFile instanceof Properties.File;
+            }
+
+            @Override
+            public @Nullable Tree visit(@Nullable Tree t, ExecutionContext ctx) {
+                if (t instanceof Yaml.Documents && sourcePathMatches(((SourceFile) t).getSourcePath(), ctx)) {
+                    t = createMergeYamlVisitor().getVisitor().visit(t, ctx);
+                } else if (t instanceof Properties.File && sourcePathMatches(((SourceFile) t).getSourcePath(), ctx)) {
+                    t = new AddProperty(propertyName(property, profile), value, comment, null)
+                            .getVisitor().visit(t, ctx);
+                }
+                return t;
+            }
+        };
+    }
+
+    private boolean sourcePathMatches(Path sourcePath, ExecutionContext ctx) {
+        List<String> expressions = pathExpressions;
+        if (expressions == null || pathExpressions.isEmpty()) {
+            // If not defined, get reasonable defaults from the execution context.
+            expressions = QuarkusExecutionContextView.view(ctx).getDefaultApplicationConfigurationPaths();
+        }
+        if (expressions.isEmpty()) {
+            return true;
+        }
+        for (String filePattern : expressions) {
+            PathMatcher pathMatcher = sourcePath.getFileSystem().getPathMatcher("glob:" + filePattern);
+            if (pathMatcher.matches(sourcePath)) {
```

Comment:
I thought we use `PathUtils.matchesGlob` in the Spring variant such that paths are reliably matched across operating systems, as `PathMatcher` is OS dependent.

Reply:
```suggestion
            if (PathUtils.matchesGlob(sourcePath)) {
```